### DISCUSSION TITLE
Excluded files in gatsby-plugin-page-creator

### DISCUSSION THREAD

**Discussion by jabrks** [gatsbyjs/gatsby#31315] [Ideas / Feature Requests]
YAML files are currently ignored by [gatsby-plugin-page-creator](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-page-creator). Would it be possible to add an option to override this behaviour? Is there any reason why this would be a bad idea?

**Comment by LekoArts** [comment]
The page creator expects valid React components with a single default export (+ optionally a named export for a graphql query). Hence YAML files can't be used -- what's the use case you're looking for here?

**Reply by jabrks** [reply]
It's quite specific, but I was looking to try and build something that generates a JSX page from a YAML file, similar to how the MDX plugin works.

**Reply by LekoArts** [reply]
Ok, I see. Then yeah, this is a feature request as https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-page-utils/src/validate-path.ts used in https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-page-creator/src/create-page-wrapper.ts doesn't accept any options. I guess it was never exposed as an option as

- no one had a need for it
- it avoided footguns for people

**Reply by jabrks** [reply]
I'd be interested in working on a PR for this, do you have any thoughts on what the API should look like?

**Reply by LekoArts** [reply]
It should be a new option on `gatsby-plugin-page-creator`, similar to the `slugify` option I added: https://github.com/gatsbyjs/gatsby/pull/28485

Let me know if you need any guidance!
https://www.gatsbyjs.com/contributing/how-to-open-a-pull-request/

**Reply by jabrks** [reply]
I've opened a PR to try and address this here: https://github.com/gatsbyjs/gatsby/pull/31375.

I'm not sure if the name I've used for the new option is descriptive enough so let me know if you can think of something better.

**Reply by LekoArts** [reply]
Thanks! I'll be on PTO for the rest of the week so I can have a look next week :)

**Reply by jabrks** [reply]
Sounds good, thanks!
